src/main/java/sk/hfa/instagram/domain/service/InstagramTokenService.java

```diff
@@ -0,0 +1,39 @@
+package sk.hfa.instagram.domain.service;
+
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+import sk.hfa.instagram.domain.InstagramToken;
+import sk.hfa.instagram.domain.repository.InstagramTokenRepository;
+import sk.hfa.instagram.domain.service.interfaces.IInstagramTokenService;
+
+import java.util.List;
+
+@Service
+public class InstagramTokenService implements IInstagramTokenService {
+
+    @Value("${hfa.instagram.token}")
+    private String token;
+
+    private final InstagramTokenRepository instagramTokenRepository;
+
+    public InstagramTokenService(InstagramTokenRepository instagramTokenRepository) {
+        this.instagramTokenRepository = instagramTokenRepository;
+    }
+
+    @Override
+    public InstagramToken findToken() {
+        List<InstagramToken> instagramToken = instagramTokenRepository.findAll();
+        if (instagramToken.isEmpty()) {
+            instagramTokenRepository.save(new InstagramToken(token, true));
+            instagramToken = instagramTokenRepository.findAll();
```

Comment:
Toto volanie do db by osm vymazal a nahradil radsej
`InstagramToken instagramToken = new InstagramToken(token, true);`
` instagramTokens = new ArrayList<>(Collections.singletonList(instagramToken));`
` instagramTokenRepository.save(instagramToken);`

Reply:
To co si navrhol asi nieje dobre riesenie lebo necheme pri kazdom vyziadani tokenu ho aj persistovat do DB nanovo